File: gcode/src/main/java/org/qw3rtrun/p3d/g/decoder/WaitReceivedDecoder.java
package org.qw3rtrun.p3d.g.decoder;

import org.apache.commons.lang3.StringUtils;
import org.qw3rtrun.p3d.g.event.WaitReceived;

import java.util.Optional;

public class WaitReceivedDecoder implements GEventDecoder<WaitReceived> {

    private static final String WAIT_PREFIX = "wait";

    @Override
    public Optional<WaitReceived> decode(String line) {
        if (line.length() != 4 || !StringUtils.startsWithIgnoreCase(line, WAIT_PREFIX)) {
            return Optional.empty();
        }
        return Optional.of(WaitReceived.getInstance());
    }
}

File: gcode/src/main/kotlin/org/qw3rtrun/p3d/g/decoder/WaitReceivedDecoder.kt
package org.qw3rtrun.p3d.g.decoder

import org.apache.commons.lang3.StringUtils
import org.qw3rtrun.p3d.g.event.WaitReceived
import java.util.Optional

class WaitReceivedDecoder : GEventDecoder<WaitReceived> {

    override fun decode(line: String): Optional<WaitReceived> {
        if (line.length != 4 || !line.startsWith(WAIT_PREFIX, ignoreCase = true)) {
            return Optional.empty()
        }
        return Optional.of(WaitReceived.getInstance())
    }

    companion object {
        private const val WAIT_PREFIX = "wait"
    }
}
